packages/cli/test/unit/generate.test.js

```diff
@@ -134,4 +134,20 @@ describe('generate', () => {
 
     expect(createLock).not.toHaveBeenCalled()
   })
+
+  test('throw error when fail-on-page-error enabled and page errors', async () => {
+    mockGetNuxt()
+    mockGetGenerator(() => ({ errors: [{ type: 'dummy' }] }))
+
+    const cmd = NuxtCommand.from(generate, ['generate', '.', '--fail-on-page-error'])
+    await expect(cmd.run()).rejects
+  })
+
+  test('do not throw an error when fail-on-page-error disabled and page errors', async () => {
+    mockGetNuxt()
+    mockGetGenerator(() => ({ errors: [{ type: 'dummy' }] }))
+
+    const cmd = NuxtCommand.from(generate, ['generate', '.'])
+    await cmd.run()
+  })
```

Comment:
I decided to test on unit level since it's a simple feature for the moment - please let me know if other e2e tests would be interesting to add.

Reply:
I think that would be more than enough. E2E test would be somehow costly.